src/couch/src/couch_users_db.erl

```diff
@@ -63,6 +64,21 @@ before_doc_update(Doc, Db, _UpdateType) ->
 save_doc(#doc{body = {Body}} = Doc) ->
     %% Support both schemes to smooth migration from legacy scheme
     Scheme = chttpd_util:get_chttpd_auth_config("password_scheme", "pbkdf2"),
+
+    % We preserve the salt value if requested (for a hashing upgrade, typically)
+    % in order to avoid conflicts if multiple nodes try to upgrade at the same time
+    % and to avoid invalidating existing session cookies (since the password did not
+    % change).
+    PreserveSalt = couch_util:get_value(?PRESERVE_SALT, Body, false),
+    Salt =
+        if
+            PreserveSalt ->
+                % use existing salt, if present.
+                couch_util:get_value(?SALT, Body, couch_uuids:random());
+            true ->
+                couch_uuids:random()
```

Comment:
Minor nit: `if true -> ... ; true -> ... end` even if correct will look confusing.  A simple `case` might be more clear instead?

Reply:
sure